### PR TITLE
fixEmptyPrintln

### LINKED ISSUE #2648: Replace hint with `print_with_newline` deletes semicolon
Given the simple source code

```rust
fn main() {
    println!("");
}
```

clippy (Version 0.0.193) creates the following error:

```
warning: using `println!("")`
 --> src/main.rs:2:5
  |
2 |     println!("");
  |     ^^^^^^^^^^^^^ help: replace it with: `println!()`
  |
  = note: #[warn(print_with_newline)] on by default
```

The "replace it with" hint falsely removes the semicolon. Note how the `^` include the semicolon but not the replacement text. This is especially bad if you use automatic fixes as provided with RLS.

Side-note: I find if pretty confusing that the newline-lint is responsible for linting this, because no newline is involved.

### PR DESCRIPTION
**Pull request by thekidxp**
fixes #2648. trying to update the message to better reflect what should be replaced

### CODE REVIEW

**Comment by thekidxp** [comment]
#2648 fixing the incorrect hint message

**Comment by mikerite** [comment]
Be careful that you don't break the case where the suggestion shouldn't have a semi-colon.

```
fn main() {
    match "a" {
        _ => println!(""),
    }
}
```
```
warning: using `println!("")`
 --> src/main.rs:3:14
  |
3 |         _ => println!(""),
  |              ^^^^^^^^^^^^ help: replace it with: `println!()`
  |
  = note: #[warn(print_with_newline)] on by default
```

**Comment by oli-obk** [comment]
huh? this changed a lot of unrelated tests. Can you make sure that you are rebased on the latest master and are running the current nightly?

You can add `fixes #2648` to the first message in this PR, so #2648 gets closed once this PR is merged

**Comment by thekidxp** [comment]
Got it, I'll take a look at the concerns. Sorry, it took me a while just figuring out how to send the pull request. I'd like to run through this again though and see if I can get it to a point where you can merge it in. I should be able to resend something later this evening.

**Comment by oli-obk** [comment]
Don't worry about it! Take all the time you need. If you have any questions, I'm also often reachable on irc. Otherwise just ask here.

**Comment by thekidxp** [comment]
Awesome! I've been loving learning rust lately and this was my first time trying to contribute to a project. Thanks for all the help. I'll reach out if I run into any more issues.

**Comment by thekidxp** [comment]
Hey, I think I finally was able to get this done. Working with github was causing me some issues so I ended up pulling down a new version of the repo. Would it be best to open a new pull request? Or just move my changes back to my fork? Also do I need a new fork or can I just push from a local branch?

**Comment by thekidxp** [comment]
got my changes for you to checkout. I'm still getting some strange issues with a few of the tests though so if possible it'd probably be good to go over those with you before anything is finalized the tests that should be updated and write.rs are the relevant changes to the ticket

**Comment by thekidxp** [comment]
I think this were the only ones that weren't line endings. I'll make sure I have latest and latest nightly, run the update script and commit again if there's anything else.

**Comment by thekidxp** [comment]
Just double checked everything and all tests are passing locally, everything is up date as far as I can tell and update-all-scripts didn't change any more files

**Comment by oli-obk** [comment]
Oooh I think I know what this is!

Please run cargo update. Maybe Travis is fetching a newer version of some markdown dependency

**Comment by thekidxp** [comment]
I've been working on this and I got it down to what looks like mostly whitespace issues. Pulling down master also won't build the tests on my local. I must have something set up wrong either with git itself or the solution. Everything is definitely up to date, latest master, and I've ran cargo update, and rustup update.

File: tests/ui/empty_line_after_outer_attribute.stderr [hunk 1]
```diff
@@ -2,7 +2,7 @@ error: Found an empty line after an outer attribute. Perhaps you forgot to add a
  --> $DIR/empty_line_after_outer_attribute.rs:5:1
   |
 5 | / #[crate_type = "lib"]
-6 | |
+6 | |
```
Comment (oli-obk) [inline tests/ui/empty_line_after_outer_attribute.stderr]: It looks to me like Travis is unhappy with these new spaces. I'll have a look later today. Sorry that you ran into this. We should've detected this on master and fixed it independently

Comment (thekidxp) [reply tests/ui/empty_line_after_outer_attribute.stderr]: No problem, I've got to mess around with more rust than I have in a while and get some experience working with travis and github that I didn't have. I appreciate that you have been patient while I worked on this. Figuring out how to work with these large sets of rust code and working with github has been good. Let me know if there's something I can look into or check on my side.

**Comment by oli-obk** [comment]
I have not been able to figure out what's going on, but the latest master + latest rustc seems to not reproduce it, so... no clue what that was.

**Pull request merged**